compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2897,6 +2897,27 @@ impl<'a> Parser<'a> {
                 }
                 self.recover_struct_comma_after_dotdot(exp_span);
                 break;
+            } else if self.token == token::DotDotDot {
+                // suggest `..v` instead of `...v`
```

Comment:
Could you move this logic into a function? To keep `parse_struct_fields` free of diagnostic logic as much as we can. :)

Reply:
Split out an function and make it shorter.